### ISSUE TITLE
Background color

### ISSUE THREAD

**Issue by miguelalvescosta** [rechsteiner/Parchment#52]
i want the background color to change based on which item is selected

**Comment by rechsteiner** [comment]
There is no built in option for defining this, but you can achieve it pretty easily by subclassing `PagingCell` and setting the background color based on the selected state. Just create your own subclass and override `setPagingItem`:

```Swift
class CustomPagingCell: PagingTitleCell {
  
  override func setPagingItem(_ pagingItem: PagingItem, selected: Bool, theme: PagingTheme) {
    super.setPagingItem(pagingItem, selected: selected, theme: theme)
    
    if selected {
      contentView.backgroundColor = .red
    } else {
      contentView.backgroundColor = .white
    }
}
```

Then set the `menuItemClass` property to `CustomPagingCell.self`:

```Swift
struct Options: PagingOptions {
  let menuItemClass: PagingCell.Type = CustomPagingCell.self
}
```

And pass that into the `PagingViewController`:

```Swift
FixedPagingViewController(viewControllers: viewControllers, options: Options())
```

This will set the background color when the selected state changes, without any animation. If you want to tween between the colors based on how much the user has scrolled, you can override `apply:layoutAttributes` instead of `setPagingItem`:

```Swift
class CustomPagingCell: PagingTitleCell {
 
  open override func apply(_ layoutAttributes: UICollectionViewLayoutAttributes) {
    super.apply(layoutAttributes)
    if let attributes = layoutAttributes as? PagingCellLayoutAttributes {
      contentView.backgroundColor = UIColor.interpolate(
        from: .white,
        to: .red,
        with: attributes.progress)
    }
  }
  
}
```

This will interpolate between the selected background color based on the scroll progress. I'm using `UIColor.interpolate` here which you can find the source code for in`UIColor+interpolation.swift`.

**Comment by miguelalvescosta** [comment]
huum thanks, one more question, oh i get click on menu cell?

**Comment by rechsteiner** [comment]
If you are using `FixedPagingViewController` you can use the `FixedPagingViewControllerDelegate` to get notified when an item was scrolled to:

```Swift
public protocol FixedPagingViewControllerDelegate : class {
  func fixedPagingViewController(fixedPagingViewController: FixedPagingViewController, willScrollToItem: ViewControllerItem, atIndex index: Int)
  func fixedPagingViewController(fixedPagingViewController: FixedPagingViewController, didScrollToItem: ViewControllerItem, atIndex index: Int)
}
```

Or if you need the actual select event you create your own subclass and override `didSelectItem`:

```Swift
class MyPagingViewController: FixedPagingViewController {
  override public func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
    super.collectionView(collectionView, didSelectItemAt: indexPath)
    print("did select")
  }
}
```

**Comment by miguelalvescosta** [comment]
class ViewController: UIViewController {

  override func viewDidLoad() {
    super.viewDidLoad()
    
    // Load each of the view controllers you want to embed
    // from the storyboard.
    let storyboard = UIStoryboard(name: "Main", bundle: nil)
    let firstViewController = storyboard.instantiateViewController(withIdentifier: "FirstViewController")
    let secondViewController = storyboard.instantiateViewController(withIdentifier: "SecondViewController")
    
    // Initialize a FixedPagingViewController and pass
    // in the view controllers.
    let pagingViewController = FixedPagingViewController(viewControllers: [
      firstViewController,
      secondViewController
    ])
    
    // Make sure you add the PagingViewController as a child view
    // controller and contrain it to the edges of the view.
    addChildViewController(pagingViewController)
    view.addSubview(pagingViewController.view)
    view.constrainToEdges(pagingViewController.view)
    pagingViewController.didMove(toParentViewController: self)

  }
}


sorry for my poor knowledge, but in a class like this how I create mine subclass and override didselectitem?

**Comment by rechsteiner** [comment]
You need to create a new class outside `ViewController` – like the `MyPagingViewController` snippet above – and replace this line:

```Swift
let pagingViewController = FixedPagingViewController(viewControllers: [
```

with this:

```Swift
let pagingViewController = MyPagingViewController(viewControllers: [
```

**Comment by miguelalvescosta** [comment]
thanks for support :)

**Comment by rechsteiner** [comment]
No problem, happy to help!

**Comment by miguelalvescosta** [comment]
hi, its me again..... can u explain me "If you are using FixedPagingViewController you can use the FixedPagingViewControllerDelegate to get notified when an item was scrolled to", i dnt know who to use delegate :/..

**Comment by rechsteiner** [comment]
No worries, the documentation for this isn't very good. You need to conform to the `FixedPagingViewControllerDelegate` protocol in you `ViewController` class. Here an example:

```Swift
extension ViewController: FixedPagingViewControllerDelegate {

  func fixedPagingViewController(fixedPagingViewController: FixedPagingViewController, willScrollToItem: ViewControllerItem, atIndex index: Int) {
    print("will scroll to item")
  }

  func fixedPagingViewController(fixedPagingViewController: FixedPagingViewController, didScrollToItem: ViewControllerItem, atIndex index: Int) {
     print("did scroll to item")
  }

}
```

Then set the `itemDelegate` property in `viewDidLoad`:

```Swift
pagingViewController.itemDelegate = self
```

**Comment by miguelalvescosta** [comment]
hi, i think i found a bug, in fixedPageViewController u have : 
  open override func em_pageViewController(_ pageViewController: EMPageViewController, didFinishScrollingFrom startingViewController: UIViewController?, destinationViewController: UIViewController, transitionSuccessful: Bool) {
    super.em_pageViewController(pageViewController, didFinishScrollingFrom: startingViewController, destinationViewController: destinationViewController, transitionSuccessful: transitionSuccessful)
       
        if let index = items.index(where: { $0.viewController == destinationViewController }) {
            itemDelegate?.fixedPagingViewController(
                fixedPagingViewController: self,
                didScrollToItem: items[index],
                atIndex: index)
        }  
    }
   
 i think u need check if transition is successful because if u do a "fake swipe" and dnt swipe to next menu tab and stay on same menu tab and didscroolToItem is activated.... in my opinion the code looks like this:

  open override func em_pageViewController(_ pageViewController: EMPageViewController, didFinishScrollingFrom startingViewController: UIViewController?, destinationViewController: UIViewController, transitionSuccessful: Bool) {
    super.em_pageViewController(pageViewController, didFinishScrollingFrom: startingViewController, destinationViewController: destinationViewController, transitionSuccessful: transitionSuccessful)
    if transitionSuccessful == true {
        
        if let index = items.index(where: { $0.viewController == destinationViewController }) {
            itemDelegate?.fixedPagingViewController(
                fixedPagingViewController: self,
                didScrollToItem: items[index],
                atIndex: index)
            }

         }
      }

**Comment by rechsteiner** [comment]
Yeah, that's definitely a bug. I also found another bug were the `willScrollItemItem` was not called at all. Fixed in #63 – will release a new version soon. Thanks for catching that 🙌

**Comment by miguelalvescosta** [comment]
u welcome :)
